daemon/cmd/daemon.go

```diff
@@ -595,6 +595,8 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 		}
 	} else if option.Config.EnableIPMasqAgent {
 		log.Fatalf("BPF ip-masq-agent requires --%s=\"true\" and --%s=\"true\"", option.Masquerade, option.EnableBPFMasquerade)
+	} else if option.Config.EnableEgressGateway {
+		log.Fatalf("Egress Gateway requires --%s=\"true\" and --%s=\"true\"", option.Masquerade, option.EnableBPFMasquerade)
```

Comment:
I think we also need to check that tunneling is enabled. It's easy to forget about that and lose 10s of minutes trying to find what's broken as I just did :sweat_smile:

Reply:
It also works when tunnel is disabled. In tunnel disabled mode, it also creates a tunnel interface only for forwarding traffic. https://github.com/cilium/cilium/pull/14723/files#diff-3bd10cddebdc5669fc7a472e02c78702a9bc59fee30d502d1cc87a84a58439d8R309-R314

Reply:
It didn't work the first time I tried, but maybe that was because I tried to use an egress IP different from the Internal IP :shrug: 

Seems to work now.